src/openrct2/actions/SetCheatAction.cpp

```diff
@@ -644,7 +644,11 @@ void SetCheatAction::RemoveAllGuests() const
                     auto peep = TryGetEntity<Guest>(peepInTrainIndex);
                     if (peep != nullptr)
                     {
-                        vehicle->mass -= peep->Mass;
+                        if ((peep->State == PeepState::OnRide && peep->RideSubState == PeepRideSubState::OnRide)
+                            || (peep->State == PeepState::LeavingRide && peep->RideSubState == PeepRideSubState::LeaveVehicle))
+                        {
+                            vehicle->mass -= peep->Mass;
```

Comment:
```suggestion
if (vehicle->mass > peep->Mass)
    vehicle->mass -= peep->Mass;
else
    vehicle->mass = 0;
```

Reply:
Or
```suggestion
                            vehicle->ApplyMass(-peep->Mass);
```

Reply:
Why though? We don't verify in any other location adding/removing.

Reply:
Yeah we should do it though. Ideally we prevent all possible underflows / overflows. What reason is there not to?

Reply:
Okay I've added a function for it.